geth-dev-node/start.sh

```diff
@@ -0,0 +1,3 @@
+geth --password /app/password.txt account new
```

Comment:
I believe this will create an account with an otherwise unknown address.  I recommend instead creating an account with a well known address.  For example, this private key is created with the empty seed and has no password:
```
{
    "id": "e1042fa7-e661-aeb8-ca86-6b9999adc120",
    "version": 3,
    "crypto": {
        "cipher": "aes-128-ctr",
        "cipherparams": {
            "iv": "50f44a53e4e11a424074063200b2e333"
        },
        "ciphertext": "190abdf9a8e4135defcd3f890a574b178f24a25fbc46f0daab3ffbb83c83a59d",
        "kdf": "pbkdf2",
        "kdfparams": {
            "c": 10240,
            "dklen": 32,
            "prf": "hmac-sha256",
            "salt": "90a1e58cfa95479588cf754dbad73f8019d2469303dd69028e8d7a2f27bf8d82"
        },
        "mac": "27a556b61e874e7f41376b4b01930caf36cee1977722f441fb73ad8cb45fba03"
    },
    "address": "00a329c0648769a73afac7f9381e08fb43dbea72",
    "name": "Development Account",
    "meta": "{\"description\":\"Never use this account outside of develoopment chain!\",\"passwordHint\":\"Password is empty string\"}"
}
```

Reply:
Ok, I'll try using that account instead.

Reply:
Note: it is possible that using a pre-created account causes problems is Geth, if so ignore this.

Reply:
I tried doing: 
```
RUN echo "" > /app/password.txt
```
in Dockerfile, and then 
```
geth --password /app/password.txt --unlock "00a329c0648769a73afac7f9381e08fb43dbea72" --mine --networkid=10000 --ws --wsapi eth,net,web3,personal --wsport 8546 --rpc --rpcapi eth,net,web3,personal,miner --rpcaddr 0.0.0.0 --targetgaslimit 6700000 --ethash.dagdir "/app/.ethash"
```
in start.sh, but it failed to unlock the account.  I also tried passing in "" instead of /app/password.txt, but it prompted me to type the password.

I'm not sure where in `genesis.json` I should be putting what you have above.  I don't think I've seen an example like that.

Reply:
The JSON file goes in a keys folder (not sure where Geth keeps them exactly, should be listed on the Geth wiki).  I don't think filename actually matters.

Reply:
Alright, I think I know what to do now, but I'm going to focus on fixing the `getTypeName` bug first, since that's higher priority.

Reply:
BTW, I think the "UnhandledPromiseRejectionWarning" messages @tinybike was seeing may have been because the account became locked after the contracts started uploading.  Once I made the Docker file unlock the account initially, those messages went away.  Now all I see is the `createMarket` bug.

Reply:
I ran geth with a permanently unlocked account, so I don't think this is the case.

Reply:
@tinybike How did you permanently unlock it?  Did you do it when starting Geth?

Reply:
Hmm, interestingly, I finally saw the `UnhandledPromiseRejectionWarning` messages just now while using Docker.

NOTE: I've made 2 changes to my setup that you may not have, @tinybike:

- I made the change in this patch by Micah for ethjs-rpc: https://github.com/MicahZoltu/ethjs-rpc/commit/1343a7a2667f1d0b96c644f18f2be1e174835229.  (I made it in `augur-core/node_modules/ethjs/src/index.js` and `augur-core/node_modules/ethjs/lib/index.js`.)

- I also added `return result;` to `augur-core/source/libraries/AbiParser.ts` right after your call to `waitForTransactionToBeSealed`.

Doing both of these can give more helpful error messages.

When running with docker, I saw these slightly more helpful messages, which say "authentication needed: password or unlock":
```
(node:2163) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: [ethjs-rpc] rpc error with payload {"id":3285766361764,"jsonrpc":"2.0","params":[{"from":"0x3ccf9547e8238c04329bf434b8b64081891c4dc0","gas":"0x5b8d80","gasPrice":"0x3b9aca00","to":"0xd400c43145602d93c1545f979a3bf279dc420d2b","data":"0x095ea7b3000000000000000000000000853f471794ca98036d3c2fbf34e7a1fa2e8137da0000000000000000000000000000000000000000000000000000000000000001"}],"method":"eth_sendTransaction"} {"code":-32000,"message":"authentication needed: password or unlock"}
(node:2163) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:2163) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: [ethjs-rpc] rpc error with payload {"id":3285766361764,"jsonrpc":"2.0","params":[{"from":"0x3ccf9547e8238c04329bf434b8b64081891c4dc0","gas":"0x5b8d80","gasPrice":"0x3b9aca00","to":"0xd400c43145602d93c1545f979a3bf279dc420d2b","data":"0x095ea7b3000000000000000000000000853f471794ca98036d3c2fbf34e7a1fa2e8137da0000000000000000000000000000000000000000000000000000000000000001"}],"method":"eth_sendTransaction"} {"code":-32000,"message":"authentication needed: password or unlock"}
```

What's odd, though, is that Docker is unlocking the account when Geth starts, and I don't typically get these errors.

Reply:
I also noticed that when I restarted my Docker container just now, I was no longer getting `UnhandledPromiseRejectionWarning`.  When I closed my laptop last night (with the Docker container was still running), I was not seeing the `UnhandledPromiseRejectionWarning` messages at all.  However, once I started running the deployment script again this morning with the Docker container that had been running all night, the messages were appearing consistently.